### PR TITLE
FSET-1068: New screens to select a list of locations and schemes that you're eligible for

### PR DESCRIPTION
**Pull request by ollyw**
(Written by HC):

Very Happy Path

WIP, current issues:

Unit / Integration Tests
Acceptance Tests

### CODE REVIEW

**Comment by henricook** [comment]
LGTM 👍

File: app/repositories/LocationSchemeRepository.scala [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  private lazy val cachedLocationSchemes =  {
+    // TODO: File needs updating with correct scheme and location data
+    val input = managed(Play.application.resourceAsStream("locations-schemes.json"))
+    val loaded = input.acquireAndGet(r => Json.parse(r).as[Locations])
+    Future.successful(loaded.locations)
+  }
+
+  def getSchemesAndLocations: Future[List[LocationSchemes]] = cachedLocationSchemes
+
+  // TODO: Needs updating with correct scheme data
```
Comment (satishkamatam) [inline app/repositories/LocationSchemeRepository.scala]: Will this be a static list in memory as is or a file ??

File: app/controllers/LocationSchemeController.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    Action.async(parse.json) { implicit request =>
+      withJsonBody[List[String]] { locationIds =>
+        locationSchemeService.updateSchemeLocations(applicationId, locationIds).map { _ => Ok }
+      }
+    }
+
+  def updateSchemes(applicationId: String): Action[JsValue] =
+    Action.async(parse.json) { implicit request =>
+      withJsonBody[List[String]] { schemeNames =>
+        locationSchemeService.updateSchemes(applicationId, schemeNames).map { _ => Ok }
```
Comment (henricook) [inline app/controllers/LocationSchemeController.scala]: The order is the order of the list

File: app/model/ProgressStatuses.scala [hunk 3]
```diff
@@ -19,7 +19,8 @@ package model
 object ProgressStatuses {
   val RegisteredProgress = "registered"
   val PersonalDetailsCompletedProgress = "personal_details_completed"
-  val SchemesAndLocationCompletedProgress = "schemes_and_locations_completed"
+  val LocationsCompletedProgress = "scheme-locations_completed"
```
Comment (henricook) [inline app/model/ProgressStatuses.scala]: :+1:

File: app/repositories/LocationSchemeRepository.scala (hunk 1 above)
Comment (henricook) [inline app/repositories/LocationSchemeRepository.scala]: Ultimately this will be a flat file, we've still got to receive this final data from Al

**Pull request merged**